component/main.jsonnet

```diff
@@ -69,16 +49,29 @@ local identityProviders = [
 
   idp {
     ldap+: {
-      ca: { name: authentication.RefName(idp.name) },
-      bindPassword: { name: authentication.RefName(idp.name) },
+      ca: { name: common.RefName(idp.name) },
+      bindPassword: {
+        name: if std.objectHas(idp, 'bindPasswordSecretRef') then params.secrets[idp.bindPasswordSecretRef] else common.RefName(idp.name),
+      },
     },
   }
   for idpname in std.objectFields(idps)
   if idps[idpname].type == 'LDAP'
+] + [
+  local idp = idps[idpname];
+  idp {
+    openID+: {
+      clientSecret: {
+        name: params.secrets[idp.openID.clientSecretRef],
```

Comment:
```suggestion
        name: idp.openID.clientSecretRef,
```

Reply:
With the latest syntax how to configured OIDC this special case basically goes away.

Reply:
yep, I added this comment before suggesting my change (this is one of the bugs that I mentioned in my review).